docs/apworld dev faq.md

```diff
@@ -0,0 +1,26 @@
+# APWorld Dev FAQ
+
+This document is meant as a reference tool to show the best known solutions to common problems when developing an apworld
+
+## My game has a restrictive start that leads to fill errors
+
+
+Hint to the Generator that an item needs to be in sphere one with local_early_items
```

Comment:
I don't know what it looks like in a markdown renderer, but this looks a little strange in this plain text form.

The `##` header is closer to the unrelated text above it, than it is to the related text below it.

I think it would make more sense to have 1 empty line between the `##` header and its corresponding text below it, and 2 empty lines above the header.

You could consider adding `---` line breaks between items too.

markdown is designed to look good in plain text as well as having a good rendered form, so I'd like to make sure it also looks good in plain text form.

Reply:
should be one line between on both sides https://google.github.io/styleguide/docguide/style.html#headings there should be breaks between code too.

Reply:
liked the `---` idea but i'm not sure if that supplanted the double newline idea or not, so i moved forward with just one empty newline between every part but adding the `---` between sections (wrapped with their own newlines)
also if you want to see the markdown render in github from the diff, you can "view file" from the overflow menu and it renders .md files

Reply:
> should be one line between on both sides https://google.github.io/styleguide/docguide/style.html#headings there should be breaks between code too.

did not see this before updating/replying, this is agreeing with my current spacings right? 
i'm not certain what you mean about breaks between code though, like change the first code block to be this?:
```
early_item_name = "Sword"

self.multiworld.local_early_items[self.player][early_item_name] = 1
```